nesta_ds_utils/file_ops.py

```diff
@@ -1,53 +1,196 @@
+    return buffer
+
+
+def _fileobj_to_df(fileobj: io.BytesIO, load_file_dir: str) -> pd.DataFrame:
+    """Convert bytes file object into DataFrame.
+
+    Args:
+        buffer (io.BytesIO): Bytes file object.
+        load_file_dir (str): Loading file name.
+
+    Returns:
+        pd.DataFrame: Dataframe converted
+    """
+    if fnmatch(load_file_dir, "*.csv"):
+        df = pd.read_csv(fileobj)
+    elif fnmatch(load_file_dir, "*.json"):
+        df = pd.read_json(fileobj)
+    elif fnmatch(load_file_dir, "*.pkl"):
+        df = pd.read_pickle(fileobj)
+    elif fnmatch(load_file_dir, "*.xml"):
+        df = pd.read_xml(fileobj)
+    else:
+        raise Exception(
+            "Automatic convertion into pd.DataFrame currently supported only for 'csv', 'json', 'pkl' and 'xml'."
+        )
+    return df
+
+
+def upload_obj(
+    obj: Union[io.BytesIO, pd.DataFrame], bucket: str, file_name_to: str, **kwargs
+):
+    """Uploads data from memory to S3 location.
+
+    Args:
+        obj (Union[io.BytesIO, pd.DataFrame]): Data to upload.
+        bucket (str): Bucket's name.
+        file_name_to (str): Path location to save data.
+
+    Raises:
+        Exception: Util supports data only as "io.BytesIO" or "pd.DataFrame".
+    """
+    s3 = boto3.client("s3")
+    if isinstance(obj, pd.DataFrame):
+        obj = _df_to_fileobj(obj, file_name_to)
+    if isinstance(obj, io.BytesIO):
+        s3.upload_fileobj(obj, bucket, file_name_to, kwargs)
+    else:
+        raise Exception('Util supports data only as "io.BytesIO" or "pd.DataFrame".')
+
+
+def download_obj(
+    bucket: str, file_name_from: str, asDataFrame: bool = False, **kwargs
+) -> Union[io.BytesIO, pd.DataFrame]:
+    """Download data to memory from S3 location.
+
+    Args:
+        bucket (str): Bucket's name.
+        file_name_from (str): Path to data in S3.
+        asDataFrame (bool, optional): If True: return the data as pd.DataFrame.
+        If False: return data as io.BytesIO. Default: False.
+
+    Returns:
+        Union[io.BytesIO, pd.DataFrame]: Donwloaded data.
+    """
+    s3 = boto3.client("s3")
+    fileobj = io.BytesIO()
+    s3.download_fileobj(bucket, file_name_from, fileobj, kwargs)
+    fileobj.seek(0)
+    return _fileobj_to_df(fileobj, file_name_from) if asDataFrame else fileobj
+
+
+def upload_file(file_name_from: str, bucket: str, file_name_to: str, **kwargs):
+    """Upload local file from disk to S3 location.
+
+    Args:
+        file_name_from (str): Path to local file.
+        bucket (str): Bucket's name.
+        file_name_to (str): Destination path in S3.
+    """
+    s3 = boto3.client("s3")
+    s3.upload_file(file_name_from, bucket, file_name_to, kwargs)
+
+
+def download_file(file_name_from: str, bucket: str, file_name_to: str, **kwargs):
+    """Download data from S3 to local file on disk.
+
+    Args:
+        file_name_from (str): Path to data in S3
+        bucket (str): Buket's name
+        file_name_to (str): Destination path to disk.
```

Comment:
Let's make sure to have consistency with naming between path, name and dir type args. I think it should be:

- path = path/to/the/file.something
- name = file.something
- dir = dir/containing/the/file/

Reply:
open to other/more sensible conventions!